qiskit_experiments/curve_analysis/composite_curve_analysis.py

```diff
@@ -385,12 +385,11 @@ def _run_analysis(
             self.plotter.set_supplementary_data(fit_red_chi=red_chi)
 
         # Create analysis results by combining all fit data
-        if total_quality == "good":
-            primary_results = self._create_analysis_results(
-                fit_data=fit_dataset, quality=total_quality, **self.options.extra.copy()
-            )
-            analysis_results.extend(primary_results)
-            self.plotter.set_supplementary_data(primary_results=primary_results)
+        primary_results = self._create_analysis_results(
+            fit_data=fit_dataset, quality=total_quality, **self.options.extra.copy()
+        )
+        analysis_results.extend(primary_results)
+        self.plotter.set_supplementary_data(primary_results=primary_results)
```

Comment:
```suggestion
        for all(fit_data.success for fit_data in fit_dataset.values()):
            primary_results = self._create_analysis_results(
                fit_data=fit_dataset, quality=total_quality, **self.options.extra.copy()
            )
            analysis_results.extend(primary_results)
            self.plotter.set_supplementary_data(primary_results=primary_results)
```
This code guarantees all fit results of sub fits exist, i.e. `fit_data.success = True` guarantees you have fit parameters in the dataset (but chi-sq can be large), otherwise fit parameter doesn't exist and code will crash when you try to pull that value from the `fit_dataset`. This is not affected by the fit quality that the experiment evaluates.
https://github.com/Qiskit/qiskit-experiments/blob/0e6153d6f4d2fc61989ec85e9e241546f9d91855/qiskit_experiments/curve_analysis/utils.py#L142-L148